controllers/controllers.go

```diff
@@ -189,3 +190,13 @@ func isValidWindowsNode(o client.Object, byoh bool) bool {
 	}
 	return true
 }
+
+// tryToMarkAsFree is called after a controller's Reconcile function returns. If the given controller finished
+// reconciling without error or requesting a requeue event, the controller is marked as free.
+// When all controllers are free, WMCO upgrades are unblocked.
```

Comment:
LIke @aravindhp  mentioned , if we caught in endless loop, should we have a timer or something so that we will allow upgrade instead of getting into deadloack ?